crates/walrus-service/src/node/metrics.rs

```diff
@@ -291,6 +291,8 @@ impl TelemetryLabel for ClientErrorKind {
             ClientErrorKind::FailedToLoadCerts(_) => "failed-to-load-certs",
             ClientErrorKind::Other(_) => "unknown",
             ClientErrorKind::StoreBlobInternal(_) => "store-blob-internal",
+            // Note: this is required due to the `#[non_exhaustive]` attribute on the enum.
+            _ => "telemetry-label-unhandled",
         }
```

Comment:
We are effectively creating work for ourselves later by doing this, as this compiler error/fix exhibits. The upsides seem negligible compared to the downsides.

`#[non_exhaustive]` seems to make it harder to augment sum-types with extra downstream functionality. 

This is a case where it is harmless to force exhaustive matching, yet this change makes it impossible to be exhaustive in the future without unnecessary toil or slippage in handling new cases, etc.

Reply:
Hmm... I guess the issue is that this implementation is outside the crate defining `ClientErrorKind` (as the `#[non_exhaustive]` doesn't have an effect in the defining crate). A way to work around this would be to define a function `label` or similar in `walrus-sdk` and then just call it here in the implementation of `TelemetryLabel`. WDYT?

But I do see the downside of the `non_exhaustive` attribute. I'm happy to keep it as it is now without adding it if you prefer.

Reply:
We could add a `.label()` method but this bleeds the requirements of the users into the implementation when they could just as well do it themselves if we avoid `non_exhaustive`. I guess I don't see the point of the `non_exhaustive` attribute except for source-compatibility, but I don't think that's high on our priority list? I'm going to go ahead and close this, but I don't have strong feelings about it. I just don't see the benefit. Feel free to reopen.